code/data_dictionary.R

```diff
@@ -86,33 +140,36 @@ adbs[adbs$State == "AE",] # when viewing ZIP, Lending data, this does indeed app
 
 # Impute Missing Fields: state --------------------------------------------------------------
 # Impute missing state by zipcode. Here, we do not check if city and zipcode match
+adbs$ImputedState = adbs$State
 cat(sprintf("%s missing states", nrow(adbs[adbs$ImputedState=='XX',])))
 XXstate_zip = adbs[(adbs$State=='XX' & !is.na(adbs$Zip)),]$Zip
 XXstate_city = adbs[(adbs$State=='XX' & !is.na(adbs$Zip)),]$City
-cat(sprintf("%s missing states, with zip entry", length(XXstate_zips)))
+cat(sprintf("%s missing states, with zip entry", length(XXstate_zip)))
 
 
-adbs$ImputedState =adbs$State
 imputed_states = c()
+imputed_states2 = c()
 missing_zips = c()
 for (zip in XXstate_zip){
+  
   state = uszips[uszips$zip==zip,]$state_id
 
   if (identical(state, character(0))){
     state = NA
     missing_zips = c(missing_zips, zip)}
   imputed_states=c(imputed_states, state)
 }
+imputed_states
 
 adbs[which(adbs$ImputedState=='XX' & !is.na(adbs$Zip)), arr.ind=TRUE]$ImputedState = imputed_states
```

Comment:
Received error: 
> Error in `[.tbl_df`(`*tmp*`, which(adbs$ImputedState == "XX" & !is.na(adbs$Zip)),  : 
  unused argument (arr.ind = TRUE)
```suggestion
adbs$ImputedState[(adbs$ImputedState=='XX' & !is.na(adbs$Zip))] = imputed_states
```